controllers/ovs_controller.go

```
@@ -0,0 +1,370 @@
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ServiceConfigReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.ServiceConfigReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	//
+	// create hash over all the different input resources to identify if any those changed
+	// and a restart/recreate is required.
+	//
+	inputHash, err := r.createHashOfInputHashes(ctx, instance, configMapVars)
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.ServiceConfigReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.ServiceConfigReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+	// TODO(slaweq): configure service (ovn controller and ovs settings)
+	instance.Status.Conditions.MarkTrue(condition.ServiceConfigReadyCondition, condition.ServiceConfigReadyMessage)
+
+	//
+	// TODO check when/if Init, Update, or Upgrade should/could be skipped
+	//
+
+	serviceLabels := map[string]string{
+		common.AppSelector: ovs.ServiceName,
+	}
+
+	// Handle service init
+	ctrlResult, err := r.reconcileInit(ctx, instance, helper)
+	if err != nil {
+		return ctrlResult, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// Handle service update
+	ctrlResult, err = r.reconcileUpdate(ctx, instance, helper)
+	if err != nil {
+		return ctrlResult, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// Handle service upgrade
+	ctrlResult, err = r.reconcileUpgrade(ctx, instance, helper)
+	if err != nil {
+		return ctrlResult, err
+	} else if (ctrlResult != ctrl.Result{}) {
+		return ctrlResult, nil
+	}
+
+	// Create additional Physical Network Attachements
+	if err := ovs.CreateAdditionalNetworks(ctx, instance, serviceLabels, r.Client); err != nil {
+		r.Log.Info(fmt.Sprintf("Failed to create additional networks: %s", err))
```

shouldn't we also return from here instead of just logging the failure so it do not move forward?

---

Done